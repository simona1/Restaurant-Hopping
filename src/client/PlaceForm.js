// @flow
import React from 'react';

type Props = {};

type State = {
  value: string,
};

class PlaceForm extends React.Component<Props, State> {
  state = {value: ''};

  _handleChange = (event: Event): void => {
    event.preventDefault();
    this.setState({value: event.target.value});
  };

  render() {
    return (
      <form onSubmit={this._handleSubmit} className="input-field">
        <i className="material-icons prefix">search</i>
        <input
          type="text"
          id="place"
          placeholder="pick a place to search for nearby establishments"
          value={this.state.value}
          onChange={this._handleChange}
        />
      </form>
    );
  }
}

export default PlaceForm;
