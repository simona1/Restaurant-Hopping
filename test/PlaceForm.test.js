import React from 'react';
import PlaceForm from '../src/client/PlaceForm.js';
import { shallow, mount, render } from 'enzyme';

describe('PlaceForm', () => {
  it('should be defined', () => {
    expect(PlaceForm).toBeDefined();
  });
  it('renders correctly', () => {
    const tree = shallow(
      <PlaceForm />
    );
    expect(tree).toMatchSnapshot();
  });

  it('should contain node with id place', () => {
    const wrapper = shallow(<PlaceForm />);
    expect(wrapper.contains('#place'));
  });
});

describe('PlaceForm', () => {
  let app;
  beforeEach( () =>  {
    app = shallow(<PlaceForm />)
  });
  it ('PlaceForm renders nested components', () => {
    expect(app.find('#place').length).toEqual(1);
    expect(app.find('form').length).toEqual(1);
  });

  it('should change state value', () => {
    const event = {
      preventDefault() {},
      target: {value: 'New World'}
    };
    const wrapper = mount(<PlaceForm />);
    const res = wrapper.find('input');
    res.simulate('change', event);
    expect(wrapper.state().value).toEqual('New World');
  });
});
