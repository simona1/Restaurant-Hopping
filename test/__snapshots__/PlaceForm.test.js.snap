// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceForm renders correctly 1`] = `
<form>
  <input
    id="place"
    onChange={[Function]}
    placeholder="type place"
    type="text"
    value=""
  />
</form>
`;
